server/apiv2/handlers/keyspace.go

```diff
@@ -0,0 +1,346 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pingcap/kvproto/pkg/keyspacepb"
+	"github.com/tikv/pd/pkg/errs"
+	"github.com/tikv/pd/server"
+	"github.com/tikv/pd/server/apiv2/middlewares"
+	"github.com/tikv/pd/server/keyspace"
+)
+
+// RegisterKeyspace register keyspace related handlers to router paths.
+func RegisterKeyspace(r *gin.RouterGroup) {
+	router := r.Group("keyspaces")
+	router.Use(middlewares.BootstrapChecker())
+	router.POST("", CreateKeyspace)
+	router.GET("", LoadAllKeyspaces)
+	router.GET("/:name", LoadKeyspace)
+	router.POST("/:name/update-config", UpdateKeyspaceConfig)
+	router.POST("/:name/enable", EnableKeyspace)
+	router.POST("/:name/disable", DisableKeyspace)
+	router.POST("/:name/archive", ArchiveKeyspace)
```

Comment:
seem they are not meet the RESTful convention?

Reply:
The last four follows POST + custom action format

Reply:
I think this standard is defined by google itself. If you think it's more expressive, we may need to update the RFC.